index.ts

```diff
@@ -70,6 +74,28 @@ export const sandbox = {
     },
   }) {
     local = await launch({})
+
+    //Load the ML model
+    //Skips if no ml.ts file is found in package using this plugin
+    const ml_path = join(cwd, ml_file)
+    let result
+    if (await exists(ml_path)) {
+      console.log(`Found ${ml_file} file, deploying ML model`)
+      result = (await import(pathToFileURL(ml_path).toString())).default
+
+      //result should be a function that returns a promise
+      if (typeof result === 'function') {
+        result = result({ node: local.url })
+      }
+
+      //wait for the returned promise to resolve
+      if (result instanceof Promise) {
+        await result
+      }
+    } else {
+      console.log(`No ${ml_file} file found, skipping ML model deployment`)
+    }
+
```

Comment:
We need to run the user-provided script in both sandbox node and real deployments. Please try implementing [`deploy.end`](https://arc.codes/docs/en/guides/plugins/deploy#deploy.end).

Reply:
Does deploy.end run for both production and sandbox environments?

Reply:
I don't know.

Reply:
Alright, no prob. I'll do some research on it.

Reply:
Hey, do you have any idea how I can get the endpoint URL from the cloud formation template? I believe I need AWS permissions or something to access the template on the stack.

Reply:
> Hey, do you have any idea how I can get the endpoint URL from the cloud formation template?

I added the following two sections to the README in order document how to read the API endpoint:

- https://github.com/nasa-gcn/architect-plugin-search#connecting-to-opensearch-from-your-application
- https://github.com/nasa-gcn/architect-plugin-search#advanced-usage-from-your-application

Both of these methods rely on Architect service discovery, which I _think_ should work even in the plugin code. If not, then you ought to be able to read the service discovery data directly from the [inventory](https://arc.codes/docs/en/guides/plugins/inventory) passed to the plugin entry points.

> I believe I need AWS permissions to access the template on the stack.

Sadly, it would take a lot of time and paperwork to get you AWS console access because we would need to get you a NASA PIV card among other things. Just try your best, and when you feel that it is ready for live testing, one of us (@dakota002, @Courey, or myself) will test and debug it in the course of code review.

Reply:
Alright then. I added the API call functionality to deploy.end(). It works in sandbox mode. I'm hoping it works in production too.